Sources/LiveKit/Broadcast/BroadcastScreenCapturer.swift

```diff
@@ -53,22 +48,32 @@ class BroadcastScreenCapturer: BufferCapturer {
             .toEncodeSafeDimensions()
 
         set(dimensions: targetDimensions)
+        return createReceiver()
+    }
 
-        let frameReader = SocketConnectionFrameReader()
-        guard let socketConnection = BroadcastServerSocketConnection(filePath: socketPath, streamDelegate: frameReader)
-        else { return false }
-        frameReader.didCapture = { pixelBuffer, rotation in
-            self.capture(pixelBuffer, rotation: rotation.toLKType())
+    private func createReceiver() -> Bool {
+        guard let socketPath = BroadcastBundleInfo.socketPath else {
+            logger.error("Bundle settings improperly configured for screen capture")
+            return false
         }
-        frameReader.didEnd = { [weak self] in
-            guard let self else { return }
-            Task {
-                try await self.stopCapture()
+        Task { [weak self] in
+            do {
+                let receiver = try await BroadcastReceiver(socketPath: socketPath)
+                logger.debug("Broadcast receiver connected")
+                self?.receiver = receiver
```

Comment:
does this have to be done within the task?  it would be better if creating the receiver and assigning it to self could be done synchronously within createReceiver

Reply:
This is a result of a design decision I made with `IPCChannel`, which `BroadcastReceiver` uses under the hood. Specifically, I designed `IPCChannel` to use the RAII pattern, making its initialization asynchronous so that the consumer always receives an open channel after initialization. This approach also allows `IPCChannel` to conform to `Sendable` without additional synchronization, as the underlying `NWConnection` is both a constant (let) and `Sendable`. I am open to feedback about this design.